pkg/kubelet/cm/dra/state/state_checkpoint.go

```diff
@@ -54,11 +55,42 @@ type ClaimInfoState struct {
 	// PodUIDs is a set of pod UIDs that reference a resource
 	PodUIDs sets.Set[string]
 
+	// ResourceHandles is a list of opaque resource data for processing by a specific kubelet plugin
+	ResourceHandles []resourcev1alpha2.ResourceHandle
+
 	// CDIDevices is a map of DriverName --> CDI devices returned by the
 	// GRPC API call NodePrepareResource
 	CDIDevices map[string][]string
 }
 
+// ClaimInfoStateWithoutResourceHandles is an old implementation of the ClaimInfoState
+type ClaimInfoStateWithoutResourceHandles struct {
+	// Name of the DRA driver
+	DriverName string
+
+	// ClassName is a resource class of the claim
+	ClassName string
+
+	// ClaimUID is an UID of the resource claim
+	ClaimUID types.UID
+
+	// ClaimName is a name of the resource claim
+	ClaimName string
+
+	// Namespace is a claim namespace
+	Namespace string
+
+	// PodUIDs is a set of pod UIDs that reference a resource
+	PodUIDs sets.Set[string]
+
+	// CDIDevices is a map of DriverName --> CDI devices returned by the
+	// GRPC API call NodePrepareResource
+	CDIDevices map[string][]string
+}
+
+// List of claim info to store in checkpoint
+type ClaimInfoStateListWithoutResourceHandles []ClaimInfoStateWithoutResourceHandles
+
```

Comment:
This should be declared next to `ClaimInfoStateList` -- either move that one here or this one there.